### PR TITLE
ENH: added accept_rate as a keyword to basinhopping

### LINKED ISSUE #8774: In `optimize.basinhopping`, the target acceptance rate should be passable as a keyword argument.

### PR DESCRIPTION
**Pull request by rocapp**
Added accept_rate as an optional keyword argument to the `optimize.basinhopping` method.

Fixes #8774

### CODE REVIEW

File: scipy/optimize/_basinhopping.py [hunk 1]
```diff
@@ -325,8 +325,8 @@ def __call__(self, **kwargs):
 
 def basinhopping(func, x0, niter=100, T=1.0, stepsize=0.5,
                  minimizer_kwargs=None, take_step=None, accept_test=None,
-                 callback=None, interval=50, disp=False, niter_success=None,
-                 seed=None):
+                 accept_rate=0.5, callback=None, interval=50, disp=False,
+                 niter_success=None, seed=None):
```
Comment (ilayn) [inline scipy/optimize/_basinhopping.py]: new keywords need to appear at the end of the signature for backwards compatibility.

File: scipy/optimize/_basinhopping.py [hunk 2]
```diff
@@ -383,6 +383,10 @@ def basinhopping(func, x0, niter=100, T=1.0, stepsize=0.5,
         other tests in order to accept the step. This can be used, for example,
         to forcefully escape from a local minimum that ``basinhopping`` is
         trapped in.
+    accept_rate : float, optional
```
Comment (rgommers) [inline scipy/optimize/_basinhopping.py]: This description needs to be at the end of the parameters list.

File: scipy/optimize/_basinhopping.py [hunk 3]
```diff
@@ -383,6 +383,10 @@ def basinhopping(func, x0, niter=100, T=1.0, stepsize=0.5,
...
+        Target acceptance rate that is used to adjust the stepsize. If the
+        current acceptance rate is greater than the target, then the stepsize
+        is increased. Otherwise, the stepsize is decreased.
```
Comment (rgommers) [inline scipy/optimize/_basinhopping.py]: minor: can you add "Default is 0.5."?

**Review by rgommers — CHANGES_REQUESTED** [review]
Thanks @rocapp. This looks like a reasonable addition to me. Just a few minor comments.

**Comment by andyfaff** [comment]
Would be superseded by #14941. Thank you for creating this PR though.

**Pull request closed without merging**